image/image.go

```diff
@@ -232,21 +309,37 @@ func bootstrapToRootDir(sto Store, model *asserts.Model, opts *Options) error {
 		}
 	}
 
+	seen := make(map[string]bool)
 	downloadedSnapsInfo := map[string]*snap.Info{}
 	var seedYaml snap.Seed
 	for _, snapName := range snaps {
-		fmt.Fprintf(Stdout, "Fetching %s\n", snapName)
-		fn, info, err := acquireSnap(sto, snapName, dlOpts)
-		if err != nil {
-			return err
+		name := local.Name(snapName)
+		if seen[name] {
+			fmt.Fprintf(Stdout, "%s already prepared, skipping\n", name)
+			continue
+		}
+
+		if name != snapName {
+			fmt.Fprintf(Stdout, "Copying %q (%s)\n", snapName, name)
+		} else {
+			fmt.Fprintf(Stdout, "Fetching %s\n", snapName)
 		}
 
-		// fetch the snap assertions too
-		err = fetchSnapAssertions(fn, info, f, db)
+		fn, info, err := acquireSnap(sto, name, dlOpts, local)
 		if err != nil {
 			return err
 		}
 
+		seen[name] = true
+
+		// if it comes from the store fetch the snap assertions too
```

Comment:
it doesn't seem very consistent, if the snap has assertions at this point then it's also in the store (there's no way atm to have assertion in the store but not the snap), unless we want to do as an optimisation but then we can do less cross checks, it would be nicer to consume the input from snap download but we aren't there yet